src/app.ts

```diff
@@ -13,14 +13,53 @@ interface TemperatureSummary {
   average: number
 }
 
+const records: Array<TemperatureReading> = []
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  records.push(...readings)
+  records.sort((a, b) => a.time.getTime() - b.time.getTime())
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const EMPTY_ARRAY = 0
+
+  //Filter elements by date and city
+  const recordsOrderByCityAndDate: TemperatureReading[] = records.filter(
+    (item) => item.time.getDate() === date.getDate() && item.city === city,
+  )
+
+  //Order records from high to lower temperature
+  const recordsOrderByTemp = [...recordsOrderByCityAndDate].sort(
```

Comment:
you could use the `recordsOrderByCityAndDate` instead of spreading the elements in a new array

Reply:
I couldn't do that because I used this array to return the response (you can see 56 and 57 lines), and if I change the order of `recordsOrderByCityAndDate` I will get an incorrect response, for this reason, I used a copy of the array.